rs/ic_os/network/mac_address/README.md

```diff
@@ -0,0 +1,69 @@
+# MAC Address
+
+Each IC-OS node must have a unique but deterministic MAC address derived from its BMC MAC address, deployment type (mainnet vs testnet), and variant type (SetupOS, HostOS, GuestOS, BoundaryGuestOS). This MAC address is then utilized to generate the node’s network configuration. To solve this, a schema has been devised.
+
+## Schema
+
+- **The first 8-bits: `6a`**
+  - Indicates an IPv6 interface.
```

Comment:
We drop the IPv4 docs here. It's correct that this is not in use, but it is technically still available if we ever wanted to do dedicated IPv4 interfaces. What do you think?

Reply:
Oh, I just realized that we have the ipv4 support in deterministic_ips, but not in mac_address:

`pub fn generate_mac_address(
    mgmt_mac: &FormattedMacAddress,
    deployment_environment: &str,
    node_type: &NodeType,
) -> Result<UnformattedMacAddress> {
    // Newline added to match behavior
    let seed = format!("{}{}\n", mgmt_mac.get(), deployment_environment);
    let vendor_part: String = hex::encode(Sha256::digest(seed)).chars().take(8).collect();
    let node_index = node_type.to_char();
    let mac = format!("6a0{}{}", node_index, vendor_part);
    UnformattedMacAddress::try_from(mac.as_str())
}`

I'll try to consolidate these this week. Do you think we should keep the ipv4 support? In that case, I'll put it back in the docs.

Reply:
I've added the IPv4 support back to the docs 👍